DLL_Project/Detours/JobGiver_GetFood.cs

```diff
@@ -71,13 +71,21 @@ protected override Job TryGiveJob( Pawn pawn )
                 }
                 if( foodSource is Building_AutomatedFactory )
                 {
+                    //Fixed the exepction from passing a Factory into the WillIngestStackCountOf function.
+                    //However, truely fixing this will require detouring FoodUtilities to get proper functionality.
                     var FS = foodSource as Building_AutomatedFactory;
-                    if( foodDef == null )
+                    foodDef = FS.BestProduct(FoodSynthesis.IsMeal, FoodSynthesis.SortMeal);
+                    if( !FS.HasEnoughResourcesInHoppersFor(foodDef) )
                     {
                         //CCL_Log.Message( string.Format( "Hopper for {0} needs filling", foodSource.ThingID ) );
                         hopperNeedsFilling = true;
                         hopper = FS.AdjacentReachableHopper( pawn );
                     }
+                    //TODO I think this is broken for factories. The code does not confirm there's material to fill the hopper available.
+                    //Not an issue with nutrient paste, since if you can't fill a nutrient paste hopper you're probably starving anyway.
+                    //But if you have a factory that makes Lavish Meals from meat. Colonists will wander around starving
+                    //trying to get meat even if there's fine meals in storage. I think anyway.
+                    //Also, current code would ignore a Fine meal factory if the Lavish meal factory has unreachable hoppers.

```

Comment:
For now I'm just using BestProduct as a band-aid to populate foodDef to prevent a null exception. I'll double check, but I was getting null exceptions and I'm pretty sure I traced it to foodDef coming back from TryFindBestFoodSourceFor as the factories' def, not the food itself. It looks like TryFindBestFoodSourceFor needs to be detoured to properly fix that.
As for the comment, I see your point and it's not an issue using BestProduct.